src/resolver.cairo

```diff
@@ -0,0 +1,82 @@
+#[starknet::contract]
+mod Resolver {
+    use core::array::SpanTrait;
+    use starknet::ContractAddress;
+    use ecdsa::check_ecdsa_signature;
+    use resolver::interface::resolver::{IResolver, IResolverDispatcher, IResolverDispatcherTrait};
+
+    #[storage]
+    struct Storage {
+        public_key: felt252,
+        uri: LegacyMap<felt252, felt252>,
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, _public_key: felt252, uri: Span<felt252>) {
+        self.public_key.write(_public_key);
+        self.store_uri(uri);
+    }
+
+    #[external(v0)]
+    impl ResolverImpl of IResolver<ContractState> {
+        fn resolve(
+            self: @ContractState, domain: Span<felt252>, field: felt252, hint: Span<felt252>
+        ) -> felt252 {
+            if hint.len() < 3 {
+                panic(self.get_uri(array!['offchain_resolving']));
+            }
+
+            let hashed_domain = self.hash_domain(domain);
+            let message_hash: felt252 = hash::LegacyHash::hash(
```

Comment:
Rereading this I can see a security issue, I suggest to add a max_validity timestamp. That way you can't use an old signature from the server to fake resolving to a previous address that is no longer valid. The max_validity has to be > to now. We can generate 1 hour valid signatures on the server side. Also, just as a good practice, I would suggest to start the hash definition by a felt explaining why this is about (so it can't be used again for something else). New hash could be:

hash( 'ccip_demo resolving', timestamp, hash(domain), field, value)

In this code I think the value is missing from the hash, it has to be included so that anyone can't just take a valid signature and replace the output by what he wants.